py/desispec/workflow/desi_proc_funcs.py

```diff
@@ -463,14 +463,21 @@ def create_desi_proc_batch_script(night, exp, cameras, jobdesc, queue, runtime=N
                     inparams.append(subparam)
         else:
             inparams = list(cmdline)        
-        for parameter in ['-q', '--queue', '--batch-opts']:
+        for parameter in ['--queue', '-q', '--batch-opts']:
```

Comment:
It looks like you're doing some complex argument processing here.  Are you quite certain that it is impossible for `argparse` to handle this?

Reply:
Long story short, I am not aware of `argparse`' ability to deal with this. I'll describe below and you can let me know if you are aware of such capabilities. These are nuissances of the desi_proc code writing a slurm script for itself, which means there are inputs given that define the script generation but should be kept for the version run on the compute nodes.

This is a special situation. Here we're trying to remove things from the input command before writing it into the script.   For example, the queue can be specified in the slurm file and is meaningless when running on the HPC node, so we are removing it. The same with batch options, which are things that the slurm script is defined with and the desi_proc running in the slurm script shouldn't try to specify. 

`argparse` does lots of things with parsing but I don't know of a way for it to remove some of its own items after it has already parsed. I also don't know a way to only get the arguments defined by the user on the command line. We can't just list the whole argparse object because we'd be providing 10-20 default value flags and make the srun commands quite long and chaotic.

Reply:
OK, here's how I would do it.

1. argparse parses the arguments.
2. You already know what the default values are.
3. Construct a completely new command line from scratch, rather than manipulating an existing command line, omitting default values (which you know what they are) and only including non-default values.

Reply:
I've considered your approach in the past. It is much more elaborate to implement than checking the inputs for a few commands and removing matching items.

What you're suggesting is what I eventually want to do, because it allows for more flexibility in the command line options we can implement. When we have things running smoothly, I am happy to revisit this.